week-3/todo-list/script.js

```diff
@@ -1,25 +1,68 @@
-function populateTodoList(todos) {
-  let list = document.getElementById("todo-list");
-  // Write your code to create todo list elements with completed and delete buttons here, all todos should display inside the "todo-list" element.
+let todos = [];
+const todoList = document.getElementById("todo-list");
+const removeAllButton = document.getElementById("remove-all-completed");
+
+function addNewTodo() {
+  let input = document.querySelector("input").value;
+  todos[todos.length] = { task: input, completed: false };
+  populateTodoList();
 }
 
-// These are the same todos that currently display in the HTML
-// You will want to remove the ones in the current HTML after you have created them using JavaScript
-let todos = [
-  { task: "Wash the dishes", completed: false },
-  { task: "Do the shopping", completed: false },
-];
+function removeCompleted() {
+  const indexesToExclude = [];
+  for (const item of todos) {
+    if (item.completed) {
+      let index = todos.indexOf(item);
+      indexesToExclude.push(index);
+      document.querySelector(".strike").remove();
+    }
+  }
+  todos = todos.filter((value, index) => !indexesToExclude.includes(index));
```

Comment:
If you were to use the todo.completed flag directly for filtering without using the array indexesToExclude as intermediate step. How would you go about doing that?